charmhelpers/contrib/network/ufw.py

```diff
@@ -236,29 +237,45 @@ def default_policy(policy='deny', direction='incoming'):
 
 
 def modify_access(src, dst='any', port=None, proto=None, action='allow',
-                  index=None):
+                  index=None, prepend=False, comment=None):
     """
     Grant access to an address or subnet
 
     :param src: address (e.g. 192.168.1.234) or subnet
                 (e.g. 192.168.1.0/24).
+    :type src: Optional[str]
     :param dst: destiny of the connection, if the machine has multiple IPs and
                 connections to only one of those have to accepted this is the
                 field has to be set.
+    :type dst: Optional[str]
     :param port: destiny port
+    :type port: Optional[int]
     :param proto: protocol (tcp or udp)
+    :type proto: Optional[str]
     :param action: `allow` or `delete`
+    :type action: str
     :param index: if different from None the rule is inserted at the given
                   `index`.
+    :type index: Optional[int]
+    :param prepend: Whether to insert the rule before all other rules matching
+                    the rule's IP type.
+    :type prepend: bool
+    :param comment: Create the rule with a comment
+    :type comment: Optional[str]
```

Comment:
TIL about `Optional`.  I've been using `Union[None,...]` but this is better and more correct.